src/pybamm/config.py

```diff
@@ -45,20 +48,125 @@ def is_running_tests():  # pragma: no cover
     return False
 
 
-def ask_user_opt_in(timeout=10):
+def get_input_or_timeout(timeout):  # pragma: no cover
     """
-    Ask the user if they want to opt in to telemetry.
+    Cross-platform input with timeout, using various methods depending on the
+    environment. Works in Jupyter notebooks, Windows, and Unix-like systems.
+
+    Args:
+        timeout (float): Timeout in seconds
+
+    Returns:
+        str | None: The user input if received before the timeout, or None if:
+            - The timeout was reached
+            - Telemetry is disabled via environment variable
+            - Running in a non-interactive environment
+            - An error occurred
+
+        The caller can distinguish between an empty response (user just pressed Enter)
+        which returns '', and a timeout/error which returns None.
+    """
+    # Check for telemetry disable flag
+    if os.getenv("PYBAMM_DISABLE_TELEMETRY", "false").lower() != "false":
+        return None
+
+    if not (sys.stdin.isatty() or is_notebook()):
+        return None
 
-    Parameters
-    ----------
-    timeout : float, optional
-        The timeout for the user to respond to the prompt. Default is 10 seconds.
+    # 1. Handling for Jupyter notebooks. This is a simplified
+    # implementation in comparison to widgets because notebooks
+    # are usually run interactively, and we don't need to worry
+    # about the event loop. The input is disabled when running
+    # tests due to the environment variable set.
+    if is_notebook():  # pragma: no cover
+        try:
+            from IPython.display import clear_output
+
+            user_input = input("Do you want to enable telemetry? (Y/n): ")
+            clear_output()
+
+            return user_input
+
+        except Exception:  # pragma: no cover
+            return None
```

Comment:
Not a major issue, but this has no timeout so people notebooks will just hang if they don't realize this popped up

Reply:
Yes, this is intentional. I opine that notebooks are meant to be used interactively by default – I expect this to be a one-time issue, since importing PyBaMM and choosing "yes" or "no" once would mean that the config gets saved and this never comes up again (until the config file gets deleted, of course).